### ISSUE TITLE
[JENKINS-70345] IssueApi is not serializable

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/warnings-ng-plugin#3103]
<p>When After running some tests in parallel in our jenkins pipeline we sometimes get the following error:</p>
<ul>
	<li> java.io.NotSerializableException: io.jenkins.plugins.analysis.core.restapi.IssueApi</li>
</ul>

<p> </p>

<p>Stacktrace:<br/>
<b>15:53:37</b>  Caused: java.io.NotSerializableException: io.jenkins.plugins.analysis.core.restapi.IssueApi*15:53:37*  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:274)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.writeKnownObject(RiverMarshaller.java:696)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:217)<b>15:53:37</b>  	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)<b>15:53:37</b>  	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)<b>15:53:37</b>  	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)<b>15:53:37</b>  	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:341)<b>15:53:37</b>  	at java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1858)<b>15:53:37</b>  	at java.base/java.util.HashMap.writeObject(HashMap.java:1412)<b>15:53:37</b>  	at org.jboss.marshalling.reflect.JDKSpecific$SerMethods.callWriteObject(JDKSpecific.java:89)<b>15:53:37</b>  	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:199)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1089)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)<b>15:53:37</b>  	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)<b>15:53:37</b>  	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)<b>15:53:37</b>  	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:341)<b>15:53:37</b>  	at java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1858)<b>15:53:37</b>  	at java.base/java.util.HashMap.writeObject(HashMap.java:1412)<b>15:53:37</b>  	at org.jboss.marshalling.reflect.JDKSpecific$SerMethods.callWriteObject(JDKSpecific.java:89)<b>15:53:37</b>  	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:199)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1089)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.AbstractObjectOutput.writeObject(AbstractObjectOutput.java:58)<b>15:53:37</b>  	at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:116)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.lambda$writeObject$1(RiverWriter.java:144)<b>15:53:37</b>  	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:331)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.writeObject(RiverWriter.java:143)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:578)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:554)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsStepContext$3.onSuccess(CpsStepContext.java:553)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsStepContext$3.onSuccess(CpsStepContext.java:547)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution$5$1.run(CpsFlowExecution.java:928)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$1.run(CpsVmExecutorService.java:38)<b>15:53:37</b>  	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)<b>15:53:37</b>  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<b>15:53:37</b>  	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:139)<b>15:53:37</b>  	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:30)<b>15:53:37</b>  	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:70)<b>15:53:37</b>  	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)<b>15:53:37</b>  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<b>15:53:37</b>  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)<b>15:53:37</b>  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)<b>15:53:37</b>  	at java.base/java.lang.Thread.run(Thread.java:829)<br/>
 </p>

<p>Could you please make it serializable</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mehmet_sap">mehmet_sap</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70345" target="_blank">IssueApi is not serializable</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: warnings-ng-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2023-09-21T15:38:41+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-09
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When After running some tests in parallel in our jenkins pipeline we sometimes get the following error:</p>
<ul>
	<li> java.io.NotSerializableException: io.jenkins.plugins.analysis.core.restapi.IssueApi</li>
</ul>

<p> </p>

<p>Stacktrace:
<b>15:53:37</b>  Caused: java.io.NotSerializableException: io.jenkins.plugins.analysis.core.restapi.IssueApi*15:53:37*  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:274)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.writeKnownObject(RiverMarshaller.java:696)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:217)<b>15:53:37</b>  	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)<b>15:53:37</b>  	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)<b>15:53:37</b>  	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)<b>15:53:37</b>  	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:341)<b>15:53:37</b>  	at java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1858)<b>15:53:37</b>  	at java.base/java.util.HashMap.writeObject(HashMap.java:1412)<b>15:53:37</b>  	at org.jboss.marshalling.reflect.JDKSpecific$SerMethods.callWriteObject(JDKSpecific.java:89)<b>15:53:37</b>  	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:199)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1089)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)<b>15:53:37</b>  	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)<b>15:53:37</b>  	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)<b>15:53:37</b>  	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:341)<b>15:53:37</b>  	at java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1858)<b>15:53:37</b>  	at java.base/java.util.HashMap.writeObject(HashMap.java:1412)<b>15:53:37</b>  	at org.jboss.marshalling.reflect.JDKSpecific$SerMethods.callWriteObject(JDKSpecific.java:89)<b>15:53:37</b>  	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:199)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1089)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)<b>15:53:37</b>  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)<b>15:53:37</b>  	at org.jboss.marshalling.AbstractObjectOutput.writeObject(AbstractObjectOutput.java:58)<b>15:53:37</b>  	at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:116)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.lambda$writeObject$1(RiverWriter.java:144)<b>15:53:37</b>  	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:331)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.writeObject(RiverWriter.java:143)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:578)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:554)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsStepContext$3.onSuccess(CpsStepContext.java:553)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsStepContext$3.onSuccess(CpsStepContext.java:547)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution$5$1.run(CpsFlowExecution.java:928)<b>15:53:37</b>  	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$1.run(CpsVmExecutorService.java:38)<b>15:53:37</b>  	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)<b>15:53:37</b>  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<b>15:53:37</b>  	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:139)<b>15:53:37</b>  	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:30)<b>15:53:37</b>  	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:70)<b>15:53:37</b>  	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)<b>15:53:37</b>  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)<b>15:53:37</b>  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)<b>15:53:37</b>  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)<b>15:53:37</b>  	at java.base/java.lang.Thread.run(Thread.java:829)
 </p>

<p>Could you please make it serializable</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70345?focusedId=433628&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-433628">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>How do you use the `IssueApi` class in your tests? This class is not intended to be saved. </p>
</pre>
</details>

</ul>
</details>

<p>How do you use the `IssueApi` class in your tests? This class is not intended to be saved. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mehmet_sap:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70345?focusedId=433637&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-433637">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi Ulli,</p>

<p> </p>

<p>in my pipeline I have 7 parallel processes which calls the method scanForIssues tool: checkStyle(pattern: "...") for different issues in different systems. I check the return value of the method scanForIssues if some issue includes the severity == "ERROR" and then I send an e-mail out if its so. To avoid this issue we have set a lock but still the sometimes the issue java.io.NotSerializableException: io.jenkins.plugins.analysis.core.restapi.IssueApi occurs.</p>

<p> </p>

<p>Code-Snippet (this method is called parallel):</p>

<p> </p>

<p>void setStageStatus(String systemName, String client) {
    // lock step for get rid of serialization issue (but still the serializable issue occurs sometimes) 
    lock(resource: "*Reporting") {
        def isUnstable = false
        def errorMessages = ""
        sleep 20 //Ensure that previous lock holder has time to exit and save state
        //Entering critical phase: A serialization of the current context will fail
        def annotatedReport = scanForIssues tool: checkStyle(pattern: "atc-checkstyle-${systemName}-${client}.xml")
        if(annotatedReport.size() &gt; 0) {
            def issues = annotatedReport.getIssues()
            issues.each { val -&gt; 
                if (val.getSeverity().toUpperCase() == "ERROR") {
                    isUnstable = true
                    errorMessages += "&lt;hr&gt;File Name: &lt;ul&gt;&lt;li&gt;${val.getFileName()}&lt;/li&gt;&lt;/ul&gt;  \nError Message: &lt;ul&gt;&lt;li&gt;${val.getMessage()}&lt;/li&gt;&lt;/ul&gt;\n\n"
                }
            }
        }</p>

<p>        if(isUnstable) {
            def emailReceivers = "&lt;some email revceivers&gt;"
            def subjectText = "<span class="error">&#91;UNSTABLE ATC Build&#93;</span>: The following System has Prio 1 OR Prio 2 ATC findings: ${systemName}"
            def bodyText = "&lt;pre&gt;ATC System ${systemName} has ATC findings which needs your attention. Please check the ATC findings in the systems ${systemName} and correct these:\n\n${errorMessages}&lt;/pre&gt;"
            emailext body: bodyText, subject: subjectText, to: emailReceivers
            unstable("System ${systemName} has ATC findings which needs your attention. Please check the ATC findings in the system ${systemName} and correct these")
        }
        //Leaving critical phase
    }
}</p>
</pre>
</details>

</ul>
</details>

<p>Hi Ulli,</p>

<p> </p>

<p>in my pipeline I have 7 parallel processes which calls the method scanForIssues tool: checkStyle(pattern: "...") for different issues in different systems. I check the return value of the method scanForIssues if some issue includes the severity == "ERROR" and then I send an e-mail out if its so. To avoid this issue we have set a lock but still the sometimes the issue java.io.NotSerializableException: io.jenkins.plugins.analysis.core.restapi.IssueApi occurs.</p>

<p> </p>

<p>Code-Snippet (this method is called parallel):</p>

<p> </p>

<p>void setStageStatus(String systemName, String client) {<br/>
    // lock step for get rid of serialization issue (but still the serializable issue occurs sometimes) <br/>
    lock(resource: "*Reporting") {<br/>
        def isUnstable = false<br/>
        def errorMessages = ""<br/>
        sleep 20 //Ensure that previous lock holder has time to exit and save state<br/>
        //Entering critical phase: A serialization of the current context will fail<br/>
        def annotatedReport = scanForIssues tool: checkStyle(pattern: "atc-checkstyle-${systemName}-${client}.xml")<br/>
        if(annotatedReport.size() > 0) {<br/>
            def issues = annotatedReport.getIssues()<br/>
            issues.each { val -> <br/>
                if (val.getSeverity().toUpperCase() == "ERROR") {<br/>
                    isUnstable = true<br/>
                    errorMessages += "<hr>File Name: <ul><li>${val.getFileName()}</li></ul>  \nError Message: <ul><li>${val.getMessage()}</li></ul>\n\n"<br/>
                }<br/>
            }<br/>
        }</p>

<p>        if(isUnstable) {<br/>
            def emailReceivers = "<some email revceivers>"<br/>
            def subjectText = "<span class="error">&#91;UNSTABLE ATC Build&#93;</span>: The following System has Prio 1 OR Prio 2 ATC findings: ${systemName}"<br/>
            def bodyText = "<pre>ATC System ${systemName} has ATC findings which needs your attention. Please check the ATC findings in the systems ${systemName} and correct these:\n\n${errorMessages}</pre>"<br/>
            emailext body: bodyText, subject: subjectText, to: emailReceivers<br/>
            unstable("System ${systemName} has ATC findings which needs your attention. Please check the ATC findings in the system ${systemName} and correct these")<br/>
        }<br/>
        //Leaving critical phase<br/>
    }<br/>
}</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70345?focusedId=433933&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-433933">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for adding some more details. I still do not understand why this happens as the API object is not even a member of a serialized class and will be created and disposed on the fly: <a href="https://github.com/jenkinsci/warnings-ng-plugin/blob/5ca5a1b9531b53f53ef7c13d5a30fe0cab36d80d/plugin/src/main/java/io/jenkins/plugins/analysis/core/model/IssuesDetail.java#L216" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/warnings-ng-plugin/blob/5ca5a1b9531b53f53ef7c13d5a30fe0cab36d80d/plugin/src/main/java/io/jenkins/plugins/analysis/core/model/IssuesDetail.java#L216</a> </p>

<p>Are you storing an instance in one of your scripts?</p>

<p>However, changing the API to make it serializable should not hurt. So feel free to submit a PR.</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>Thanks for adding some more details. I still do not understand why this happens as the API object is not even a member of a serialized class and will be created and disposed on the fly: <a href="https://github.com/jenkinsci/warnings-ng-plugin/blob/5ca5a1b9531b53f53ef7c13d5a30fe0cab36d80d/plugin/src/main/java/io/jenkins/plugins/analysis/core/model/IssuesDetail.java#L216" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/warnings-ng-plugin/blob/5ca5a1b9531b53f53ef7c13d5a30fe0cab36d80d/plugin/src/main/java/io/jenkins/plugins/analysis/core/model/IssuesDetail.java#L216</a> </p>

<p>Are you storing an instance in one of your scripts?</p>

<p>However, changing the API to make it serializable should not hurt. So feel free to submit a PR.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mehmet_sap:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70345?focusedId=433934&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-433934">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The script stores the instance in the variables "annotatedReport" &amp; "issues" for a short period of time.</p>

<p>Due to the parallel processing it might happen that the jenkins persists the current pipeline state at any time when one of the parallel threads finishes. </p>
</pre>
</details>

</ul>
</details>

<p>The script stores the instance in the variables "annotatedReport" & "issues" for a short period of time.</p>

<p>Due to the parallel processing it might happen that the jenkins persists the current pipeline state at any time when one of the parallel threads finishes. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70345?focusedId=433936&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-433936">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote>

<p>The script stores the instance in the variables "annotatedReport" &amp; "issues" for a short period of time.</p>

<p>Due to the parallel processing it might happen that the jenkins persists the current pipeline state at any time when one of the parallel threads finishes. </p>
</blockquote>

<p>Yes, but these instances are serializable. I wonder why an instance of <tt>io.jenkins.plugins.analysis.core.restapi.IssueApi</tt> is problematic, these instances are generated on the fly to provide access to the remote API.</p>
</pre>
</details>

</ul>
</details>

<blockquote>

<p>The script stores the instance in the variables "annotatedReport" & "issues" for a short period of time.</p>

<p>Due to the parallel processing it might happen that the jenkins persists the current pipeline state at any time when one of the parallel threads finishes. </p>
</blockquote>

<p>Yes, but these instances are serializable. I wonder why an instance of <tt>io.jenkins.plugins.analysis.core.restapi.IssueApi</tt> is problematic, these instances are generated on the fly to provide access to the remote API.</p>
